### PR TITLE
hbd clean up

### PR DESCRIPTION
**Pull request by okhlif**
### Description
Hbd clean up to support three mode for 10bit samples
   - EB_8_BIT_MD    = 8bit mode decision
   - EB_10_BIT_MD   = 10bit mode decision **(default)**
   - EB_DUAL_BIT_MD = Auto: 8bit & 10bit mode decision

Add 8bit IFS path in 10bit MD
### Author
@okhlif 
### Type of change
Cleanup
Enhancement
### Tests and performance
EB_10_BIT_MD  & EB_8_BIT_MD unchanged
Add IFS 8 bit under EB_DUAL_BIT_MD
~ 2.8% speed gain for 10bit 360p
~ BDR over 7 clips 360p-10bit -0.11% (PSNR) -0.04% (SSIM)
~ Memory increase ~1%

### CODE REVIEW

**Comment by okhlif** [comment]
@joelsole I made changes to support 3 modes. Please let me know your feedback
     * 0 = 8bit mode decision
     * 1 = 10bit mode decision
     * 2 = Auto: 8bit & 10bit mode decision

**Comment by joelsole** [comment]
Interesting -  I wonder about the benefit of introducing IFS_8BIT_MD because the reference frame might be the 8-bit or the 10-bit version depending on the part of MD being executed: what are the pros and cons?

There will be some conflicts with #769. We'll solve those when we get there.

**Comment by okhlif** [comment]
> Interesting - I wonder about the benefit of introducing IFS_8BIT_MD because the reference frame might be the 8-bit or the 10-bit version depending on the part of MD being executed: what are the pros and cons?
> 
> There will be some conflicts with #769. We'll solve those when we get there.

We always have the reference frame and input frame in both 8-bit and 10-bit. For example IFS, even the whole MD path is 10-bit, IFS can still reference 8-bit and use 8-bit input. I need to update lambda update in this case ( not yet done) 
This can bring better BDRare/speed trad-off especially the memory overhead is ~1%

**Comment by kylophone** [comment]
> @joelsole I made changes to support 3 modes. Please let me know your feedback
> * 0 = 8bit mode decision
> * 1 = 10bit mode decision
> * 2 = Auto: 8bit & 10bit mode decision

Maybe use an `enum` for these.

**Comment by okhlif** [comment]
@joelsole @chkngit  .Can you please review the changes

File: Source/Lib/Common/Codec/EbEncDecProcess.c [hunk 1]
```diff
@@ -1684,7 +1684,9 @@ void* enc_dec_kernel(void *input_ptr)
             segmentRowIndex = segment_index / segments_ptr->segment_band_count;
             segmentBandIndex = segment_index - segmentRowIndex * segments_ptr->segment_band_count;
             segmentBandSize = (segments_ptr->lcu_band_count * (segmentBandIndex + 1) + segments_ptr->segment_band_count - 1) / segments_ptr->segment_band_count;
-
+//#if HBD_CLEAN_UP // context_ptr
```
Comment (hassount) [inline Source/Lib/Common/Codec/EbEncDecProcess.c]: still needed ?

**Comment by hassount** [comment]
@okhlif 
This LGTM except for comment that might need to be removed.
This PR is up next as soon as it's ready

File: Source/Lib/Common/Codec/EbEncDecProcess.c (hunk 1 above)
Comment (okhlif) [reply Source/Lib/Common/Codec/EbEncDecProcess.c]: Removed thank you

**Comment by okhlif** [comment]
> @okhlif
> This LGTM except for comment that might need to be removed.
> This PR is up next as soon as it's ready

Thank you, comment addressed. Ready to go form my end.

**Pull request merged**